Add details to attendees modal

StevenXL opened the pull request:
I have added the Class Level Title to the attendee modal that is opened when an organizer clicks on a student. Aside from the specs and from the new `Gemfile.lock`, the changes are as follows:

1. `rsvp.rb` model has two new instance methods - `#level_title` and `#level`.
2. `event.rb` model has one new instance method - `#levels`.
3. `attendee_detail.hbs` template has a new section for the level's title.

lilliealbert commented:
Hi @StevenXL! Thanks for working on this. As a first step, can you squash all your commits into one? 

Here's a small guide about doing so if it helps: http://railsbridge.github.io/bridge_troll/

StevenXL commented:
@lilliealbert Yes I will do that. Please give me a few minutes. I made a mistake when rebasing and need to figure out what I did and how to fix it. At the moment it is listing @tjgrathwell as the author of the commit. I'll fix this and report back.

StevenXL commented:
@lilliealbert OK - please take a look now and let me know if any changes you'd recommend. I've fixed the rebasing issue, and the entire pull request is now one commit.

tjgrathwell commented:
Hey @StevenXL, I think the tests for Event#level and Rsvp#level probably aren't needed. They just add a lot of lines to the file and repeat static information from `course.rb`, such that anytime the text for those courses change someone would need to go edit these mostly unrelated tests.

The test for Rsvp#level_title seems sufficient to prove that the level title can be retrieved for a user. It might also be nice to have a test asserting that the level title actually appears in the student info modal, but there might not be any tests around that modal yet, so it might be a little harder to write.

StevenXL commented:
@tjgrathwell Thanks for looking over the pull request.

I can replace the static text in `Event#levels` with `Course::RAILS.levels` to make the test less brittle. The factory always assumes a Rails course for the event via `course_id Course::RAILS.id`

I can do something similar for `Rsvp#level` since the factory always creates an Rsvp with a `class_level` attribute set to `2`. 

I'll definitely replace the tests for `Rsvp#level_title` with something more dynamic, but let me know if the changes I suggested for `Rsvp#level` and `Event#levels` change your mind, or if I should just delete those. I'm good either way.

I'm going to take some time to try and figure out how to test the modal.

StevenXL commented:
@tjgrathwell 

I've added a test suite for the attendee detail modal - you can find it in the `spec/features` directory.

I also deleted the tests for `Event#levels` and `Rsvp#level`, but left untouched the `Rsvp#level_title'.

Let me know if there are other changes you'd recommend.

tjgrathwell commented:
Looks good, thanks!
